ops/model.py

```diff
@@ -3579,6 +3580,16 @@ def secret_info_get(
         id = list(info_dicts)[0]  # Juju returns dict of {secret_id: {info}}
         return SecretInfo.from_dict(id, typing.cast(Dict[str, Any], info_dicts[id]))
 
+    @contextlib.contextmanager
+    def _secrets_in_temp_files(self, content: Dict[str, str]) -> Generator[List[str], None, None]:
+        rv: List[str] = []
+        with contextlib.ExitStack() as temp_files:
+            for k, v in content.items():
+                temp = temp_files.enter_context(tempfile.NamedTemporaryFile(buffering=0))
```

Comment:
@tonyandrewmeyer, may I get your opinion on a couple of style choices:

Would you prefer explicit default values here?

- `mode="w+b"`
- `delete=True, delete_on_close=True`

Which do you prefer?
- `encoding=None, buffering=0` and `write(bytes)` or
- `encoding="utf-8"` and `write(str); flush()` ?

Reply:
> Would you prefer explicit default values here?
> 
> * `mode="w+b"`
> * `delete=True, delete_on_close=True`

No, I don't think there's any reason to pass values the same as the default ones.

> Which do you prefer?
> 
> * `encoding=None, buffering=0` and `write(bytes)` or
> * `encoding="utf-8"` and `write(str); flush()` ?

I think the former, but not strongly, so go with whichever you prefer.